src/utils/dts.ts

```diff
@@ -64,7 +69,29 @@ export function extractDeclarations(
       const imports = findStaticImports(contents);
       const exports = findExports(contents);
       const typeExports = findTypeExports(contents);
-      for (const spec of [...exports, ...typeExports, ...imports]) {
+      const dynamicImports = findDynamicImports(contents).map(
+        (dynamicImport) => {
+          let specifier: string | undefined;
+          try {
+            const value = JSON.parse(dynamicImport.expression);
```

Comment:
A simpler option might be to take a substring, but I wasn't sure if there might a case with an actual dynamic import where the expression is not a string literal but it still passes the `RELATIVE_RE` test.